float64 sorted_cx
float64 sorted_cy
float64 sorted_width
float64 sorted_height
